jdaviz/core/helpers.py

```diff
@@ -457,19 +433,26 @@ def get_data(self, data_label=None, cls=None, subset_to_apply=None, statistic=No
         data = self.app.data_collection[data_label]
 
         if not cls:
-            if len(data.shape) == 2 and self.app.config == "specviz2d":
+            if 'Trace' in data.meta:
+                cls = None
+            elif len(data.shape) == 2 and self.app.config == "specviz2d":
```

Comment:
Could have used `data.ndim` instead of calculating `len(data.shape)` multiple times.